### PR TITLE
fix: i18n locale resolution (spurious es/fr 404s; short locales served English)

### PR DESCRIPTION
**Pull request by kilbot**
## What & why
Reported from production console: `GET …/src/translations/es/wp-admin-landing-shared.json 404` (and `fr`, etc.).

Two bugs in the i18n config, confirmed by reproducing the actual network requests per locale:
1. **Spurious 404 every load.** i18next's default `load: 'all'` requested *both* the full locale `es_ES` (exists → 200) *and* the base language `es` (no such dir → 404). Page worked, but logged a 404 each load.
2. **Short-locale users served English.** A site whose WP locale is a language-only code (`es`, `fr`) requested only that → 404 → silent English fallback instead of the user's language.

## Fix
- `load: 'currentOnly'` — request only the exact locale (our locale files are key-complete per `lint-translations`, so no region→base fallback layer is needed).
- `loadPath` is now a function that maps the requested code to an available directory via `resolveLocaleDir` (short `es` → `es_ES`, `pt` → `pt_BR`, `zh` → `zh_CN`; unknown → passthrough → 404 → bundled English).
- Extracted to a pure, unit-tested `src/shared/locale-resolve.mjs` (`tests/locale-resolve.test.mjs`).

## Verified
Reproduced before/after with a headless harness — every locale now makes exactly **one** request to the correct dir; short codes resolve to their region; the `es`/`fr` 404s are gone.

Please review before merge — I'm not self-merging these.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Improved locale resolution to correctly handle exact language codes, hyphenated forms, and region variants with consistent directory selection.
* **Performance**
  * Optimized translation loading to request only the currently active locale, reducing extra companion requests.
* **Tests**
  * Added automated coverage for locale/region resolution behavior, including edge cases and unknown locale pass-through.

### CODE REVIEW

File: src/shared/locale-resolve.mjs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ *
+ * @param {string} lng
+ * @returns {string}
+ */
+export function resolveLocaleDir(lng) {
+  if (!lng) return lng;
+  if (LOCALE_DIRS.includes(lng)) return lng;
+  const base = lng.split(/[-_]/)[0].toLowerCase();
+  if (LOCALE_DIRS.includes(base)) return base;
+  return LOCALE_DIRS.find((dir) => dir.toLowerCase().split('_')[0] === base) ?? lng;
```
[the remark this answers is not in the document]
Comment (kilbot) [reply src/shared/locale-resolve.mjs]: Fixed in 0d61057 — `resolveLocaleDir` now normalizes `-`→`_` and matches an exact supported directory (case-insensitive) before falling back to the base language, so `zh-TW`→`zh_TW`, `es-MX`→`es_MX`, `pt-PT`→`pt_PT`. Added those multi-region hyphen cases to the test (the earlier `fr-FR` case passed only because `fr` has a single region, so it didn't catch this). Thanks.
